requirements.txt

```diff
@@ -41,8 +41,8 @@ sphinx==8.1.2
 sphinx_rtd_theme==3.0.1
 
 # Requirements for automated linting
-ruff==0.6.3
-pre-commit==3.7.1
+ruff==0.7.0
+pre-commit==4.0.1
```

Comment:
items need to be added to the change log in changes for developers.
Can you please also include changes in #17260, the changes to the pre-commit YML file.
Please also add the suggest new hooks: check-illegal-windows-names and maybe the fixed requirements-txt-fixer